pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,168 @@
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "agent.buildkite.com:aud": "sts.amazonaws.com"
+                },
+                "StringLike": {
+                    "agent.buildkite.com:sub": "organization\:ORGANIZATION_SLUG\:pipeline:PIPELINE_SLUG\:ref\:REF\:commit\:BUILD_COMMIT\:step\:STEP_KEY"
+                },
+                "IpAddress": {
+                    "aws:SourceIp": [
+                        "AGENT_PUBLIC_IP_ONE",
+                        "AGENT_PUBLIC_IP_TWO"
+                    ]
+                }
+
+            }
+        }
+    ]
+}
+```
+
+Update the following sections:
+
+- In the Principal, replace `AWS_ACCOUNT_ID`  with your account ID.
+- In the Condition on the subject, replace:
+
+    * `ORGANIZATION_SLUG` with your Buildkite Organization
+    * `PIPELINE_SLUG` with your Pipeline
+    * `REF` - this is commonly replaced with `refs/heads/main` to enforce only the main branch using the IAM role  or `refs/tags/*` for only tagged releases able to deploy, or a wildcard if we want all branches to use it.
+    * `BUILD_COMMIT` - this is commonly replaced with a wildcard `*`
+    * `STEP_KEY` - this is commonly replaced with a wildcard `*`
+
+- In the condition on the IP Address, replace `AGENT_PUBLIC_IP_ONE` and `AGENT_PUBLIC_IP_TWO` with a list of your Agent IP addresses.
+
+Expanded example Trust Policy:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "agent.buildkite.com:aud": "sts.amazonaws.com"
+                },
+                "StringLike": {
+                    "agent.buildkite.com:sub": [
+                        "organization\:example-org\:pipeline\:example-pipeline\:ref\:refs/heads/main\:*",
+                        "organization\:example-org\:pipeline\:example-pipeline\:ref\:refs/tags/*:*"
+                    ]
+                },
+                "IpAddress": {
+                    "aws:SourceIp": [
+                        "192.0.2.0",
+                        "198.51.100.0"
+                    ]
+                }
+
+            }
+        }
+    ]
+}
+```
+
+## Step 3: Configure your IAM role with AWS actions
+
+Add inline or managed IAM policies to the role to allow whatever IAM permissions your pipeline needs. Common examples are permissions to read secrets from SSM and push images to ECR, but there are no limitations here.
```

Comment:
These are the AWS docs for the inline or managed  IAM policies that need to be attached to the role to set it up to interact with the customers AWS account. 

https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html


```suggestion
Add an inline or managed IAM policy to the role to allow the IAM role to perform any actions your pipeline needs. Common examples are permissions to read secrets from SSM and push images to ECR, but this entirely depends on the purpose of your Pipeline.
```

I didn't add the link in there, but that's slightly better phrasing of what this step is doing